tests/apollo/util/skvbc.py

```diff
@@ -262,6 +289,41 @@ async def fill_and_wait_for_checkpoint(
                 expected_checkpoint_num=lambda ecn: ecn == checkpoint_before + num_of_checkpoints_to_add,
                 verify_checkpoint_persistency=verify_checkpoint_persistency,
                 assert_state_transfer_not_started=assert_state_transfer_not_started)
+    
+    async def run_concurrent_txns_and_wait_for_checkpoint(
+            self, initial_nodes,
+            write_run_duration=10,
+            verify_checkpoint_persistency=True,
+            assert_state_transfer_not_started=True,
+            without_clients=None):
+        """
+        A helper function used by tests to run txns for a duration and then
+        checkpoint it.
+        If persistency flag is set the nodes are then stopped and restarted
+        to ensure the checkpoint data was persisted.
+        filling concurrent to speed up tests
+        """
+        with log.start_action(action_type="run_concurrent_txns_and_wait_for_checkpoint"):
+            clients = self.bft_network.get_all_clients()
+            async def write_random_kv(client):
+                key = self.random_key()
+                val = self.random_value()
+                reply = await self.send_write_kv_set(client,[(key, val)])
+                assert reply.success
+            t_end = time.time() + write_run_duration
+
+            with trio.move_on_after(write_run_duration+CLIENT_REQ_GRACEFUL_END_TIMER):
+                while time.time() < t_end:
+                    async with trio.open_nursery() as nursery:
```

Comment:
Can we call send_concurrent_ops() though that is driven by number of operations which can be provided?

Reply:
no celine here we are not fixing the number of operations but running the client req for a given period of time

Reply:
@ananpal ,  I think @cjn might be right. we might have missed something here... it seems like send_concurrent_ops might be good for what we need? Just need to add an option to insert num_ops as null value and insert a duration instead?

Reply:
@glevkovich I feel that changing this function for our need doesn't look very straightforward and further it will complicate the existing code so it's better to use a new function.

Reply:
@ananpal 
How does it complicate the code?
send_concurrent_ops is used in 3-4 places, so any refactoring is not a big deal.

- replace 2nd parameter num_ops with a callable function as a parameter:
`send_concurrent_ops(self, exit_predicate, ...`

- replace
`while sent < num_ops:`
with
`while not exit_predicate():`

That's all, all you need is to define the functions (lambda) in the caller, 1 liner.
I don't see any complications.
Code reuse is important...

Reply:
done